### PR TITLE
fix: class based state issues 

### PR DESCRIPTION
**Pull request by TGTGamer**

### CODE REVIEW

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -1,34 +1,31 @@
 import './App.css'
 import { Component } from 'react'
 import React from 'react'
-import PieRechartComponent from './components/pie.rechart'
-import PieVictoryComponent from './components/pie.victory'
+import { PieData, PieRechartComponent } from './components/pie.rechart'
+import { PieVictoryComponent } from './components/pie.victory'
```
Comment (TGTGamer) [inline frontend/src/App.tsx:5]: Need to import the exposed components

File: frontend/src/App.tsx [hunk 2]
```diff
@@ -1,34 +1,31 @@
...
+
 
-type PieData = {
-    name: string
-    value: number
-  }[]
```
Comment (TGTGamer) [inline frontend/src/App.tsx:10]: Not needed as you are going to import it from the class file to reduce repeated declarations and possible issues

File: frontend/src/App.tsx [hunk 3]
```diff
@@ -1,34 +1,31 @@
...
-    window.runtime.EventsOn("cpu_usage", (cpu_usage: { avg:number }) => {
-        this.setState({usage: cpu_usage.avg})});
+        window.runtime.EventsOn("cpu_usage", (cpu_usage: { avg: number }) => {
+            this.state = ({ usage: cpu_usage.avg })
+        });
 
-      const {usage} = this.state
-      const pieData = [
+        const { usage } = this.state
+        const pieData: PieData[] = [
```
Comment (TGTGamer) [inline frontend/src/App.tsx:19]: Add type definition to reduce errors

File: frontend/src/components/pie.rechart.tsx [hunk 4]
```diff
@@ -1,16 +1,25 @@
 import React, { Component } from "react"
 import { PieChart, Pie, Cell, ResponsiveContainer } from 'recharts'
 
-type PieData = {
+export type PieData = {
```
Comment (TGTGamer) [inline frontend/src/components/pie.rechart.tsx:4]: Export the declared type so you can reuse

File: frontend/src/components/pie.rechart.tsx [hunk 5]
```diff
@@ -1,16 +1,25 @@
...
   name: string
   value: number
-}[]
+}
+
+interface PieDataProps {
+  pieData: PieData[]
+}
+
+interface States { }
```
Comment (TGTGamer) [inline frontend/src/components/pie.rechart.tsx:13]: Define the class props and states

File: frontend/src/components/pie.rechart.tsx [hunk 6]
```diff
@@ -1,16 +1,25 @@
...
 
-class PieRechartComponent extends React.Component<PieData> {
+export class PieRechartComponent extends React.Component<PieDataProps, States> {
   COLORS = ['#F26D22', 'transparent']
-  pieData = [...this.props]
 
-  render(){
+  constructor(props: PieDataProps) {
```
Comment (TGTGamer) [inline frontend/src/components/pie.rechart.tsx:18]: construct the class using the props to allow them to be passed

File: frontend/src/components/pie.rechart.tsx [hunk 7]
```diff
@@ -1,16 +1,25 @@
...
+    super(props)
```
Comment (TGTGamer) [inline frontend/src/components/pie.rechart.tsx:19]: pass the props to the React.Component class to be used

File: frontend/src/components/pie.rechart.tsx [hunk 8]
```diff
@@ -26,7 +35,7 @@ class PieRechartComponent extends React.Component<PieData> {
               {usage}
             </text> */}
             <Pie
-              data={this.pieData}
+              data={this.props.pieData}
```
Comment (TGTGamer) [inline frontend/src/components/pie.rechart.tsx:38]: Reference the props correctly

File: frontend/src/components/pie.rechart.tsx [hunk 9]
```diff
@@ -38,7 +47,7 @@ class PieRechartComponent extends React.Component<PieData> {
               cornerRadius={10}
             >
               {
-                this.pieData.map((_entry: any, index: number) => (
+                this.props.pieData.map((_entry: any, index: number) => (
```
Comment (TGTGamer) [inline frontend/src/components/pie.rechart.tsx:50]: Reference the props correctly

File: frontend/src/components/pie.rechart.tsx [hunk 10]
```diff
@@ -48,6 +57,4 @@ class PieRechartComponent extends React.Component<PieData> {
       </>
     )
   }
-}
-
-export default PieRechartComponent
```
Comment (TGTGamer) [inline frontend/src/components/pie.rechart.tsx:53]: Exporting defaults is a bad habit and should only be used for modules. You should instead export the individual elements to ensure consistency with naming conventions

**Review by BenoitBotton — COMMENTED** [review]
thanks @TGTGamer 
I learned a good lesson

**Pull request merged**